src/card.js

```diff
@@ -0,0 +1,142 @@
+    button.setAttribute("aria-expanded", "false");
+    button.type = "button";
+    button.innerText = "more info";
+    button.addEventListener("click", (e) => {
+        const thisID = e.target.id.replace("_button", "");
+        loader.classList.add("loader"); // in case of server lag / can use timeout to simulate
+        //setTimeout(() => { // todo remove - just to see it :)
+        const storage = JSON.parse(sessionStorage.getItem(thisID)); //date.id
+        if (storage !== null && (new Date()).getTime() - (new Date(storage.dateTime)).getTime() <= 120000) { // under two mins or none get from sessionStorage
+            setCurrencyInfo(storage, thisID);
+        }
+        else {
+            onInfo(thisID, thisID); //date.id
+        }
+        loader.classList.remove("loader");
+
+        cardSubDiv.classList.toggle("show");
+        //}, 700);
+    });
+    const loader = createElementWrapper("div", ["col-6"]);
+    btnDiv.append(button, loader);
+    const cardSubDiv = createElementWrapper("div", ["card", "collapse", "justify-content-between", "align-items-center", "mt-3"], collapsCardId);
+    const carsSubBodyDiv = createElementWrapper("div", ["card-body", "row", "subCard"]);
+    const imgDiv = createElementWrapper("div", ["col-4"]);
+    //html image element ea bit diferent
+    const img = createElementWrapper("img", ["img-fluid", "rounded-start"], colDiv.id + "_img");
+    img.alt = data.name;
+    img.src = LOAD_IMAGE; // takes time to load image so adding default loading png
+    // todo add loader image till loaded 
+    imgDiv.append(img);
+    const priceWrapperDiv = createElementWrapper("div", ["col-8"]);
+    const priceInnerDiv = createElementWrapper("div", ["float-end"]);
+    const usdDiv = createElementWrapper("div", ["card-text"], colDiv.id + "_usd");
+    const euroDiv = createElementWrapper("div", ["card-text"], colDiv.id + "_euro");
+    const shekelDiv = createElementWrapper("div", ["card-text"], colDiv.id + "_shekel");
+    priceInnerDiv.append(usdDiv, euroDiv, shekelDiv);
+    priceWrapperDiv.append(priceInnerDiv);
+    carsSubBodyDiv.append(imgDiv, priceWrapperDiv);
+    cardSubDiv.append(carsSubBodyDiv);
+    cardDiv.append(rowDiv, coinName, btnDiv, cardSubDiv);
+    colDiv.append(cardDiv);
+    return colDiv;
+}
+
+
+
+async function onInfo(coinId, thisID) {
+    const coinInfo = await getCoinInfo(coinId)
+    // todo add validation
+    // consider interface for that object 
+    const coinInfoStorage = new CoinInfoStorage(
+        new Date(),
+        coinInfo.image.small,
+        coinInfo.market_data.current_price.usd,
+        coinInfo.market_data.current_price.eur,
+        coinInfo.market_data.current_price.ils
+    )
+    setCurrencyInfo(coinInfoStorage, thisID)
+    sessionStorage.setItem(coinId, JSON.stringify(coinInfoStorage)) //new Date()-dd //120000
+}
+
+async function getCoinInfo(coinId) {
+    if (coinId === "") return "id..."
```

Comment:
you should throw some error here

and also can be written

`if(!coinId) throw new Error()`

OR 

be more specific
typeof